File: students/students/src/MON.Models/StudentModels/InternationalMobilityModel.cs
namespace MON.Models.StudentModels
{
    using System;
    using System.Collections.Generic;

    public class InternationalMobilityModel
    {
        public int? Id { get; set; }
        public string Project { get; set; }
        public string ReceivingInstitution { get; set; }
        public string MainObjectives { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public int PersonId { get; set; }
        public int CountryId { get; set; }
        public IEnumerable<DocumentViewModel> Documents { get; set; }
        public int? InstitutionId { get; set; }
        public short SchoolYear { get; set; }
        public string SchoolYearName { get; set; }
    }
}

File: students/students/src/MON.SPA/src/models/internationalMobilityModel.js
import Constants from '@/common/constants.js';
import { DocumentModel } from '@/models/documentModel.js';

export class InternationalMobilityModel {
    constructor(obj = {}, moment = {}) {
        this.id = obj.id || null;
        this.project = obj.project || null;
        this.receivingInstitution = obj.receivingInstitution || null;
        this.mainObjectives = obj.mainObjectives || null;
        this.countryId = obj.countryId;
        this.fromDate  = (obj.fromDate && moment) ? moment(obj.fromDate).format(Constants.DATEPICKER_FORMAT) : '';
        this.toDate  = (obj.toDate && moment) ? moment(obj.toDate).format(Constants.DATEPICKER_FORMAT) : '';
        this.personId = obj.personId || null;
        this.institutionId = obj.institutionId;
        this.schoolYear = obj.schoolYear;
        this.schoolYearName = obj.schoolYearName;
        this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    }
}
